OpenEdX/AppDelegate.swift

```diff
@@ -42,24 +45,37 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                 )
             }
             configureDeepLinkServices(launchOptions: launchOptions)
+            
+            if config.firebase.enabled {
+                FirebaseApp.configure()
+                if config.firebase.cloudMessagingEnabled {
+                    let pushManager = Container.shared.resolve(PushNotificationsManager.self)
+                    Messaging.messaging().delegate = pushManager
+                    UNUserNotificationCenter.current().delegate = pushManager
+                    UIApplication.shared.registerForRemoteNotifications()
```

Comment:
@volodymyr-chekyrta This way `didRegisterForRemoteNotificationsWithDeviceToken` delegate method will fired only if `cloudMessagingEnabled==true`. If FCM is disabled then all other providers will unable to register device token.

Reply:
Makes sense 👍 
I've updated it for any provider ✅ 
```
let pushManager = Container.shared.resolve(PushNotificationsManager.self)

if config.firebase.enabled {
    FirebaseApp.configure()
    if config.firebase.cloudMessagingEnabled {
        Messaging.messaging().delegate = pushManager
        UNUserNotificationCenter.current().delegate = pushManager
    }
}

if pushManager?.hasProviders == true {
    UIApplication.shared.registerForRemoteNotifications()
}
```